### ISSUE TITLE
Applocker log analysis generates a lot of records

### ISSUE THREAD

**Issue by jgstew** [strawgate/C3-Protect#27]
**Location:** https://github.com/strawgate/C3-Protect/blob/master/Analyses/Whitelisting%20-%20Applocker%20-%20Logs%20-%20Windows.bes

Because this relevance includes the "time generated" then every log entry is a new record returned by the analysis that is unique for every computer and event. This will significantly grow the BFEnterprise database and is problematic for any large deployments.

Using something like this would just give the unique event types:

     unique values of ( ( (if (event id of it = 8006) then ("Warned: ") else ("Blocked: ")) of it ) & (preceding text of last " was " of description of it) ) of records ((integers in(item 0 of it + item 1 of it - 1,maximum of (item 0 of it + item 1 of it - 500;item 1 of it))) of (record count of it, oldest record number of it)) whose (event id of it = 8006 or event id of it = 8007) of event log "Microsoft-Windows-AppLocker/MSI and Script"

This would give the count, which would often be unique per endpoint, but less verbose than the current state:

    (multiplicity of it as string & it) of unique values of ( ( (if (event id of it = 8006) then ("Warned: ") else ("Blocked: ")) of it ) & (preceding text of last " was " of description of it) ) of records ((integers in(item 0 of it + item 1 of it - 1,maximum of (item 0 of it + item 1 of it - 500;item 1 of it))) of (record count of it, oldest record number of it)) whose (event id of it = 8006 or event id of it = 8007) of event log "Microsoft-Windows-AppLocker/MSI and Script"

------

### Related:
- https://bigfix.me/analysis/details/2998311
- https://bigfix.me/relevance/details/3018048
- https://bigfix.me/relevance/details/3018047

**Comment by jgstew** [comment]
I have often done similar things to this in the past, but I'm starting to rethink this approach.

I think the best of both worlds is to have analyses that are much more verbose and specific that only a very small group of machines would report into for diagnostics, and potentially have machines automatically switch for a time period if there are issues detected so they can be diagnosed easier, and potentially allow IT or even users to trigger deeper diagnostics. Gathering overly specific data everywhere when you don't need it is less useful, especially when a lot can be inferred by looking at the less verbose data across all devices.

**Comment by strawgate** [comment]
I agree -- these analyses are separated out from the others into their own "Logs" analyses for this reason. The original intention was to help someone responding to a ticket so having the time generated was useful but I do like the idea of shifting it to a summary overview using relevance like you indicated.
